### PR TITLE
Add `helmify` subcommand

### PR DESCRIPTION
**Pull request by ThorstenHans**
This PR adds the `helmify` subcommand. Which allows users to generate Helm charts for a Spin App.

The command is added as subcommand to `scaffold` and re-uses the flags defined on `scaffold`.

## Usage

```
spin kube scaffold helmify -f ttl.sh/hello-spinkube:24h \
  --memory-request 64Mi \
  --memory-limit 128Mi \
  --cpu-request 100m \
  --cpu-limit 200m \
  --replicas 4
```

### CODE REVIEW

File: pkg/cmd/helmify.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+		values["runtimeConfig"] = base64.StdEncoding.EncodeToString(raw)
+	}
+	return values, nil
+}
+
+func init() {
+	// overwriting the default value of a flag defined on the parrent comand does not work
+	// that's why I went for a PreRunE
+	helmifyCmd.Flags().StringVarP(&scaffoldOpts.output, "out", "o", defaultChartOutput, "Path for writing the Helm chart to")
+	scaffoldCmd.AddCommand(helmifyCmd)
```
Comment (bacongobbler) [inline pkg/cmd/helmify.go:143]: I think it'd be better if we make this `spin kube helmify` rather than `spin kube scaffold helmify`. While both `scaffold` and `helmify` perform the same rough amount of work (prepare a Spin application to be deployed to Kubernetes), the use cases for both are quite distinct, and I think they should be treated as such.

By tying this command as a subcommand of `scaffold`, every feature flag that gets introduced to `scaffold` must now be added to `helmify`. There will be cases in the future where there will be a feature flag that will only work with `scaffold` but not `helmify`, and it'd be confusing for users to see that feature flag available in `helmify`'s help output because of the way this command has been architected.

File: go.mod [hunk 2]
```diff
@@ -12,21 +12,24 @@ require (
 	github.com/spf13/pflag v1.0.5
 	github.com/spinkube/spin-operator v0.1.0
 	github.com/stretchr/testify v1.9.0
-	k8s.io/api v0.29.2
-	k8s.io/apimachinery v0.29.2
+	helm.sh/helm/v3 v3.14.3
+	k8s.io/api v0.29.3
```
Comment (bacongobbler) [inline go.mod:16]: Why is this included if it isn't a direct dependency of the plugin?

File: go.mod [hunk 3]
```diff
@@ -12,21 +12,24 @@ require (
...
+	k8s.io/apimachinery v0.29.3
```
Comment (bacongobbler) [inline go.mod:17]: Why is this included if it isn't a direct dependency of the plugin?

**Review by bacongobbler — CHANGES_REQUESTED** [review]
Thank you for the contribution. Would you mind taking some time to write down some documentation in the README?

A PR like this was a concern of mine when the `scaffold` command was introduced to the plugin. Every company has their own tools and their own mechanisms for deploying assets to Kubernetes. By adding `scaffold`, I was concerned that we've now opened the floodgates and we will now need to maintain commands for other third-party projects like Helm, Kustomize, cdk8s, \<insert obscure template language here>... How do you envision us handling that maintenance burden over time? Do you think it'd make sense to split out `helmify` as a separate CLI plugin?

I've marked this as "request changes" because there are a few things that should be addressed before merging this PR, but I think it'd be worth addressing the larger question of whether it makes sense to maintain this as a separate plugin before proceeding.

**Comment by bacongobbler** [comment]
Alternatively, what makes this command different from `helm create && spin kube scaffold -o templates`?

**Comment by rajatjindal** [comment]
> you think it'd make sense to split out `helmify` as a separate CLI plugin

there is also https://github.com/arttor/helmify (which we use in spin-operator I believe).

File: go.mod (hunk 2 above)
Comment (ThorstenHans) [reply go.mod:16]: If you're talking about the `helm.sh/helm/v3` dependency, it's added because I decided to use `chart` and `chartutil` from helm itself instead of rolling everything manually

File: go.mod (hunk 3 above)
Comment (ThorstenHans) [reply go.mod:17]: what do you mean by "this" `k8s.io/apimachinery`? As part of the PR I added `helm` as a dependency maybe I've to `go mod tidy` it again?

**Comment by ThorstenHans** [comment]
@bacongobbler I was also uncertain if this should be added to `kube scaffold` or if it should become a custom `plugin`. Obviously, there are many ways on how to treat k8s deployment manifests.

IMO we could address the majority of scenarios with being able to scaffold plain k8s manifests and helm charts. I went for a subcommand of `scaffold` because it allowed me to re-use most parts of the CLI interface.

I'm happy to move the command around.

**Comment by ThorstenHans** [comment]
I don't see a helm starter chart as an equivalent solution for this. 

- How could we distribute the starter chart to a broad audience 
- Users have to alter the `values.yaml` and provide all the details of their app 

As addition to the 2nd point, `spin kube scaffold` provides a way more guided experience

**Comment by bacongobbler** [comment]
Using @rajatjindal's suggestion, I was able to generate a Helm chart with the following:

```
$ wget https://github.com/arttor/helmify/releases/download/v0.4.11/helmify_Linux_x86_64.tar.gz
$ tar xf helmify_Linux_x86_64.tar.gz
$ spin kube scaffold -f bacongobbler/myapp:1.0.0 | ./helmify mychart
$ cat mychart/templates/myapp.yaml
apiVersion: core.spinoperator.dev/v1alpha1
kind: SpinApp
metadata:
  name: {{ include "mychart.fullname" . }}-myapp
  labels:
  {{- include "mychart.labels" . | nindent 4 }}
spec:
  executor: containerd-shim-spin
  image: bacongobbler/myapp:1.0.0
  replicas: 2
```

It appears to write a Helm chart, and it accepts the exact same arguments as `spin kube scaffold`. For example:

```
><> spin kube scaffold -f bacongobbler/myapp:1.0.0 --replicas 3 --autoscaler hpa --cpu-limit 100m --memory-limit 128Mi | ./helmify
```

What do you think about suggesting this tool instead?

In principle, I prefer that we lean on the existing community for solutions wherever possible. Unless there is a very clear reason we need to provide a solution for folks, I'd prefer that we focus on the tooling that matters most to us, which in our case is the relationship between a Kubernetes client (the `kube plugin`), the Kubernetes API, and spin-operator.

File: pkg/cmd/resources/hpa.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  metrics:
+  - type: Resource
+    resource:
+      name: cpu
+      target:
+        type: Utilization
+        averageUtilization: {{ .Values.targetCpuUtilizationPercentage }}
+  - type: Resource
+    resource:
+     name: memory
```
Comment (bacongobbler) [inline pkg/cmd/resources/hpa.yaml:28]: This line needs to be indented one more space. I just discovered this bug and filed an issue: https://github.com/spinkube/spin-plugin-kube/issues/68

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Comment by bacongobbler** [comment]
> * How could we distribute the starter chart to a broad audience

Starter chart packs can be shared as git repositories. For example, if we were to package and distribute one here:

```
git clone git@github.com:spinkube/spin-helm-starter-pack
helm create --starter spin-helm-starter-pack my-spin-app
```

**Comment by bacongobbler** [comment]
Closing; `helmify` can take `spin kube scaffold` as input and output a Helm chart.
